syntax = "proto3";
package ytadmin;

message ServiceIdentity {
  string name = 1;
  string instance_id = 2;
  string host = 3;
  string version = 4;
}

message PushHealthRequest {
  ServiceIdentity identity = 1;
  bool healthy = 2;
  map<string, string> checks = 3;
  map<string, double> metrics = 4;
  string ts_iso = 5;
}

message PushEffConfRequest {
  ServiceIdentity identity = 1;
  map<string, string> config = 2;
  repeated string redacted_keys = 3;
  string config_hash = 4;
}

message PushAck {
  bool ok = 1;
  string message = 2;
}

message IdentityRequest {}
service AppProbe {
  rpc GetIdentity (IdentityRequest) returns (ServiceIdentity);
}

service AdminIngest {
  rpc PushHealth (PushHealthRequest) returns (PushAck);
  rpc PushEffConf (PushEffConfRequest) returns (PushAck);
}